pkg/vpcmodel/groupingSelfLoop.go

```diff
@@ -0,0 +1,335 @@
+package vpcmodel
+
+import (
+	"strings"
+)
+
+// extends grouping by considering self loops as don't care https://github.com/np-guard/vpc-network-config-analyzer/issues/98
+// e.g. a => b,c   b => a, c   and   c => a,b   is actually a clique a,b,c => a,b,c
+// a => b,c, b => c can be presented in one line as  a,b => b,c
+
+// After the basic grouping, which is of worst time complexity O(n^2), we optimize grouping treating self loops as don't care.
+// Intuitively, we check if two GroupedConnLine can be merged treating self loops as don't care
+// 1. groupsToBeMerged find couples of GroupedConnLine that should be merged using the alg below
+// mergeSelfLoops merges the groupsToBeMerged:
+// 2. It creates sets of GroupedConnLine s.t. each set should be merged
+//			note that the "should be merged" is an equivalence relation
+// 3. It merges them
+
+// alg: GroupedConnLine to be merged:
+// GroupedConnLine whose distance is an empty set should be merged;
+// the claim below guarantees that the result is coherent
+//
+// The distance between two GroupedConnLine is defined as following:
+// Let l_1 be a line with source s_1 and dest d_1 and let l_2 be a line with source s_2 and dest d_2.
+// l_1 / l_2 is the subnets in d_1 that are not in d_2 minus the single subnet in s_1 if |s_1| = 1
+//
+// The distance between lines l_1 and l_2 is l_1 / l_2 union l_2 / l_2
+//
+// claim: if the distance between line l_1 and l_2 is empty and
+//		 the distance between lines l_2 and l_3 is zero
+//		 then so is the distance between l_1 and l_3
```

Comment:
be consistent with distance term `zero` or `empty`

Reply:
right. fixed